src/client.rs

```diff
@@ -106,54 +110,102 @@ impl Client {
+        }
+        let url = self.parse_url(track.download_url.as_ref().unwrap());
+        let mut response = self.http_client.get(url).send().await?;
+        // Follow the redirect just this once.
+        if let Some(header) = response.headers().get(reqwest::header::LOCATION).cloned() {
+            let url = Url::parse(header.to_str()?).unwrap();
+            response = self.http_client.get(url).send().await?;
+        }
+        let download = response.bytes_stream();
+        // Convert the stream into an futures::io::AsyncRead
+        // but first we must convert the reqwest::Error into a futures::io::Error
+        let download = download.map_err(
+            |e| futures::io::Error::new(futures::io::ErrorKind::Other, e))
+            .into_async_read();
+        // Convert the futures::io::AsyncRead into a tokio::io::AsyncRead
+        let mut download = download.compat();
+        let bytes = tokio::io::copy(&mut download, &mut writer).await?;
+        Ok(bytes)
+    }
+
+    /// Starts streaming the track provided in the tracks `stream_url` to the `writer`
+    /// if the track is streamable via the API.
+    ///
+    /// Returns Ok if the track was retreived and written successfully, an error otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use soundcloud::Client;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///   let client = Client::new(env!("SOUNDCLOUD_CLIENT_ID"));
+    ///   let path = Path::new("hi.mp3");
+    ///   let track = client.tracks().id(263801976).get().await.unwrap();
+    ///   let mut outfile = tokio::fs::File::create(path).await.unwrap();
+    ///   let num_bytes = client.stream(&track, &mut outfile).await.unwrap();
+    ///   assert!(num_bytes > 0);
+    /// }
+    /// ```
+    pub async fn stream<W: AsyncWrite + Unpin>(&self, track: &Track, mut writer: W) -> Result<u64> {
+        if !track.streamable || !track.stream_url.is_some() {
+            return Err(Error::TrackNotStreamable);
+        }
+        let url = self.parse_url(track.stream_url.as_ref().unwrap());
+        let mut response = self.http_client.get(url).send().await?;
+        // Follow the redirect just this once.
+        if let Some(header) = response.headers().get(reqwest::header::LOCATION).cloned() {
+            let url = Url::parse(header.to_str()?).unwrap();
+            response = self.http_client.get(url).send().await?;
+        }
+        let stream = response.bytes_stream();
+        // Convert the stream into an futures::io::AsyncRead
+        // but first we must convert the reqwest::Error into a futures::io::Error
+        let stream = stream.map_err(
+            |e| futures::io::Error::new(futures::io::ErrorKind::Other, e))
+            .into_async_read();
+        // Convert the futures::io::AsyncRead into a tokio::io::AsyncRead
+        let mut stream = stream.compat();
+        let bytes = tokio::io::copy(&mut stream, &mut writer).await?;
```

Comment:
Shouldn't this use `futures::io::copy` so we don't expose tokio in our api?
If so, the tokio and tokio-util dependencies should also be marked as dev-dependencies.

Reply:
I've reverted back to using ```futures::io::copy``` as requested.

My initial motivation behind the change to ```tokio::io::copy``` was to allow the caller to pass in ```tokio::fs::File```.

Using futures the only other way I know is to pass in ```futures::io::AllowStdIo::new(std::fs::File::create("file.mp3")?)``` which blocks the event loop and doesn't take advantage of the tokio runtime.

Reply:
Wouldn't the caller be able to use the same tokio-util methods as you've used in the implementation before?
So a user of this lib should still be able to use `tokio::fs::File`

Reply:
No, at least not easily.
Here's the doc for `tokio_util::compat::FuturesAsyncReadCompatExt` trait previously used:

> Extension trait that allows converting a type implementing futures_io::AsyncRead to implement tokio::io::AsyncRead

There is a similar trait for converting `future_io::AsyncWrite`.

It won't convert the other way `tokio::io::AsyncWrite` -> `futures_io::AsyncWrite` which is what would be needed for the caller to pass in `tokio::fs::File` because currently the writer is bounded by `futures::io::AsyncWrite`.

Reply:
I've looked into the docs of tokio-util, wouldn't `tokio_util::compat::Tokio02AsyncWriteCompatExt` be what the caller could use?
It converts from `tokio::io::AsyncWrite` to `futures_io::AsyncWrite`.

Reply:
You're right and I should've read the docs. Thanks for showing me that! Awesome, so we are good to go. Do you need anything else for this PR?

Reply:
You're welcome.
I don't think so, I'll test it and if everything works as expected I'll merge it.
Thanks for your contribution!